apps/files/src/mixins.js

```diff
@@ -100,6 +100,16 @@ export default {
         })
       })
     },
+    compareIds (x, y) {
+      if (!isNaN(x)) { // OC10 autoincrement id
+        return parseInt(x) === parseInt(y)
+      } else if (atob(y).split(':').length === 2) { // OC10: https://github.com/cs3org/reva/blob/e7830e2f752a05a081178ed26c384ced983b8fde/internal/http/services/owncloud/ocdav/ocdav.go#L169-L175
```

Comment:
if this is something we'll do more in the future for extracting the id, maybe we need a separate function for just extraction ?

something like `{x, y, z, ...} = extractCS3IdParts(id)` (not sure how many parts it has)

Reply:
now thinking, maybe this should me done on SDK level. The SDK is here to abstract away whether Webdav, OCS or CS3 is being used to avoid having API-specific logic in the frontend code.

@DeepDiver1975 thoughts ?

Reply:
I'm not entirely sure this is going to stay this way, perhaps @butonic can shed a bit more light, as the only place I found this requirement is alone in the reva code

Reply:
@butonic any comments here ?

if we don't know let's add a "TODO: move to common location or library" then